src/runners/alert_processor.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+import json
+import uuid
+
+from .config import ALERTS_TABLE, DATABASE
+from .helpers import db, log
+
+CORRELATION_PERIOD = -60
+
+
+def get_correlation_id(ctx, alert):
+    try:
+        actor = alert['ACTOR']
+        object = alert['OBJECT']
+        action = alert['ACTION']
+        time = alert['EVENT_TIME']
+    except Exception as e:
+        log.error(f"Alert missing a required field: {e.args[0]}", e)
+        return uuid.uuid4().hex
+
+    # select the most recent alert which matches the correlation logic
+
+    query = f"""select * from {ALERTS_TABLE}
+    where alert:ACTOR = '{actor}'
+    and (alert:OBJECT = '{object}' or alert:ACTION = '{action}')
+    and correlation_ID is not null
+    and event_time > dateadd(minutes, {CORRELATION_PERIOD}, '{time}')
+    order by event_time desc
+    limit 1
+    """
+
+    try:
+        match = ctx.cursor().execute(query).fetchall()
+    except Exception as e:
+        log.error("Failed unexpectedly while getting correlation matches", e)
```

Comment:
Fixed; we assign match to an empty list in the exception handler, which avoids the unassigned variable reference issue, and then since match doesn't have a len > 0, we get correlation_id = uuid.uuid4().hex